hydrolib/core/io/bc/models.py

```diff
@@ -220,23 +271,177 @@ class T3D(ForcingBase):
     function: Literal["t3d"] = "t3d"
 
     offset: float = Field(0.0, alias="offset")
+    """float: All values in the table are increased by the offset (after multiplication by factor). Defaults to 0.0."""
+
     factor: float = Field(1.0, alias="factor")
+    """float: All values in the table are multiplied with the factor. Defaults to 1.0."""
 
-    verticalpositions: List[float] = Field(alias="verticalPositions")
-    verticalinterpolation: VerticalInterpolation = Field(alias="verticalInterpolation")
-    verticalpositiontype: VerticalPositionType = Field(alias="verticalPositionType")
+    vertpositions: List[float] = Field(alias="vertPositions")
+    """List[float]: The specification of the vertical positions."""
+
+    vertinterpolation: VerticalInterpolation = Field(
+        VerticalInterpolation.linear.value, alias="vertInterpolation"
+    )
+    """VerticalInterpolation: The type of vertical interpolation. Defaults to linear."""
+
+    vertpositiontype: VerticalPositionType = Field(alias="vertPositionType")
+    """VerticalPositionType: The vertical position type of the verticalpositions values."""
+
+    timeinterpolation: TimeInterpolation = Field(
+        TimeInterpolation.linear.value, alias="timeInterpolation"
+    )
+    """TimeInterpolation: The type of time interpolation. Defaults to linear."""
+
+    _key_renaming_root_validator = get_key_renaming_root_validator(
+        {
+            "timeinterpolation": ["time_interpolation"],
+            "vertpositions": ["vertical_position_specification"],
+            "vertinterpolation": ["vertical_interpolation"],
+            "vertpositiontype": ["vertical_position_type"],
+            "vertpositionindex": ["vertical_position"],
+        }
+    )
 
     _split_to_list = get_split_string_on_delimiter_validator(
-        "verticalpositions",
+        "vertpositions",
     )
 
     _verticalinterpolation_validator = get_enum_validator(
-        "verticalinterpolation", enum=VerticalInterpolation
+        "vertinterpolation", enum=VerticalInterpolation
     )
     _verticalpositiontype_validator = get_enum_validator(
-        "verticalpositiontype", enum=VerticalPositionType
+        "vertpositiontype", enum=VerticalPositionType
+    )
+    _timeinterpolation_validator = get_enum_validator(
+        "timeinterpolation", enum=TimeInterpolation
     )
 
+    @root_validator(pre=True)
+    def _validate_quantityunitpairs(cls, values: Dict) -> Dict:
+        super()._validate_quantityunitpair(values)
+
+        quantityunitpairs = values["quantityunitpair"]
+
+        T3D._validate_that_first_unit_is_time_and_has_no_verticalposition(
+            quantityunitpairs
+        )
+
+        verticalpositions = values.get("vertpositions")
+        if verticalpositions is None:
+            raise ValueError("vertpositions is not provided")
```

Comment:
Convention is to use the lower camel case names of the keywords: `vertPositions` & `vertPositionIndex`

Reply:
Done